### PR TITLE
Have setup template add a license

### LINKED ISSUE #29: License of this repo should be MIT or CC0
Otherwise it can't be used in projects with a more permissive license.

### PR DESCRIPTION
**Pull request by ReinderVosDeWael**
- Resolves #29 
- Addresses parts of #35 

This PR replaces this repo's license with the MIT license, and adds a license selection to the template setup. Currently, this includes LGPL2.1, MIT, and GPL3. Additional ones can easily be added in the new licenses directory.

### CODE REVIEW

File: setup_template.py [hunk 1]
```diff
@@ -36,31 +129,36 @@ def main():
...
     )
     input("Press enter to continue...")
 
     # Replace the template values
-    for file in pl.Path(dir_repo).glob("**/*"):
+    for file in pl.Path(DIR_REPO).glob("**/*"):
         if (
             file.is_file()
-            and not file.name == file_self.name
+            and not file.name == __file__
```
Comment (nx10) [inline setup_template.py:149]: Pretty sure I tried that and it did not work reliably. (I think when working dir != project dir.

File: setup_template.py [hunk 2]
```diff
@@ -70,16 +168,18 @@ def main():
...
-    dir_module = dir_repo / "src" / "APP_NAME"
+    replace_license(repo_license)
+
+    dir_module = DIR_REPO / "src" / "APP_NAME"
     if dir_module.exists():
         dir_module.rename(dir_module.parent / module_name)
 
     # Remove this file
-    print(f"Removing {file_self.name}")
+    print(f"Removing {__file__}")
```
Comment (nx10) [inline setup_template.py:182]: This also prints the full path this way

**Comment by nx10** [comment]
GitHub also has a REST API for licenses: https://docs.github.com/en/rest/licenses

File: setup_template.py (hunk 2 above)
Comment (ReinderVosDeWael) [reply setup_template.py]: Replaced with a catch-all statement as more files are removed now.

File: setup_template.py (hunk 1 above)
Comment (ReinderVosDeWael) [reply setup_template.py]: Replaced with `setup_template.py`, no need to get fancy.

**Comment by ReinderVosDeWael** [comment]
Revamped to use the Github API. New licenses can now be added in the settings file.

@nx10 , the `setup_template.py` was becoming quite illegible with all the code required to make the licenses work well taking up half the space. What do you think of expanding into a `setup` directory as I've done here? Would also go well with further complexity that may arise from automating the pre-commit install.

**Comment by nx10** [comment]
Yeah it's probably better this way, the intention at first was to keep the minimum python version this supported as low as possible and avoid the module path. (To be able to use system python everywhere). But yeah probably easier to maintain this way.

**Pull request merged**